hack/local-e2e-common.sh

```diff
@@ -0,0 +1,474 @@
+    if command -v oc &> /dev/null; then
+        if KUBECONFIG="$KUBECONFIG" oc cluster-info &> /dev/null; then
+            echo "✓ Cluster is accessible via oc"
+            CLUSTER_ACCESSIBLE=true
+        fi
+    fi
+
+    if [ "$CLUSTER_ACCESSIBLE" = "false" ] && command -v kubectl &> /dev/null; then
+        if KUBECONFIG="$KUBECONFIG" kubectl cluster-info &> /dev/null; then
+            echo "✓ Cluster is accessible via kubectl"
+            CLUSTER_ACCESSIBLE=true
+        fi
+    fi
+
+    if [ "$CLUSTER_ACCESSIBLE" = "false" ]; then
+        if ! command -v oc &> /dev/null && ! command -v kubectl &> /dev/null; then
+            echo "Error: Neither oc nor kubectl found. Please install one of them." >&2
+        else
+            echo "Error: Cannot connect to cluster using KUBECONFIG=$KUBECONFIG" >&2
+            echo "Please verify:" >&2
+            echo "  1. The kubeconfig file is valid" >&2
+            echo "  2. The cluster is accessible" >&2
+            echo "  3. Your credentials are valid" >&2
+        fi
+        exit 1
+    fi
+}
+
+# Setup HIVE_IMAGE: auto-detect, prompt, or use existing value
+# Must export the variable so e2e-common.sh and child processes can access it
+function setup_hive_image() {
+    if [ -z "$HIVE_IMAGE" ]; then
+        echo "HIVE_IMAGE not set, attempting to get latest tag from quay.io..."
+        if ! command -v curl &> /dev/null || ! command -v jq &> /dev/null; then
+            echo "Warning: curl or jq not found. Cannot auto-detect Hive image tag." >&2
+            echo "Please install curl and jq, or set HIVE_IMAGE environment variable." >&2
+            echo "Please enter the Hive image to use (e.g. quay.io/my-user/hive:latest):"
+            read -r HIVE_IMAGE
+            if [ -z "$HIVE_IMAGE" ]; then
+                echo "Error: HIVE_IMAGE cannot be empty" >&2
+                exit 1
+            fi
+        else
+            HIVE_IMAGE_REPO="${HIVE_IMAGE_REPO:-redhat-user-workloads/crt-redhat-acm-tenant/hive-operator/hive}"
+            LATEST_TAG=$(curl -sk "https://quay.io/api/v1/repository/${HIVE_IMAGE_REPO}/tag/?page=1&limit=100" | \
+                jq -r '.tags | map(select(.name | startswith("hive-on-push") and endswith("-index"))) | sort_by(.start_ts) | reverse | .[0].name // empty' 2>/dev/null)
+            if [ -n "$LATEST_TAG" ] && [ "$LATEST_TAG" != "null" ] && [ "$LATEST_TAG" != "" ]; then
+                HIVE_IMAGE="quay.io/${HIVE_IMAGE_REPO}:${LATEST_TAG}"
+                echo "Auto-detected HIVE_IMAGE: $HIVE_IMAGE"
+            else
+                echo "Failed to auto-detect Hive image tag"
+                echo "Please enter the Hive image to use (e.g. quay.io/my-user/hive:latest):"
+                read -r HIVE_IMAGE
+                if [ -z "$HIVE_IMAGE" ]; then
+                    echo "Error: HIVE_IMAGE cannot be empty" >&2
+                    exit 1
+                fi
+            fi
+        fi
+    fi
+    export HIVE_IMAGE="$HIVE_IMAGE"
```

Comment:
nit: I believe this can just be spelled
```suggestion
    export HIVE_IMAGE
```